datanode/snapshot/config.go

```diff
@@ -0,0 +1,24 @@
+package snapshot
+
+import "code.vegaprotocol.io/vega/datanode/config/encoding"
+
+type Config struct {
+	RemoveSnapshotsOnStartup encoding.Bool `long:"remove-snapshops-on-startup" description:"if true the nodes snapshot directory will be emptied at startup"`
+
+	BlockInterval int64         `long:"block-interval" description:"the block interval between creation of snapshots, 0 if snapshot create should be disabled (default: 0)"`
+	Publish       encoding.Bool `long:"publish" description:"set to true if this node should publish snapshot data"`
+	PanicOnError  encoding.Bool `long:"panic-on-error" description:"true if service should panic on snapshot creation error"`
+
+	DatabaseSnapshotsPath string `long:"database-snapshot-path" description:"the snapshots path relative the database working directory"`
+}
+
+// NewDefaultConfig creates an instance of the package specific configuration, given a
+// pointer to a logger instance to be used for logging within the package.
+func NewDefaultConfig() Config {
+	return Config{
+		DatabaseSnapshotsPath:    "/snapshots",
+		RemoveSnapshotsOnStartup: false,
+		BlockInterval:            1000, // Roughly every 15 mins at 1 block per second
```

Comment:
I'm not really sure how/where these deltas will be published, but will it could it cause issues if different data-nodes have `BlockInterval` set to different values?

Reply:
yes, very much, ideally this will need to a network parameter or similar to make sure they are in sync with other datanodes and core.  I've put this in the TBD column for now and left in config. But we will need to address this.

Reply:
let's use the existing network parameter that we use for the core for now? and then we'll figure if it needs to be changed? Having core and datanode take snapshots at the same time everytime shall make things simpler at first.